### PR TITLE
Complete thread abort support for the new EH

### PR DESCRIPTION
**Pull request by janvorli**
This change adds all the necessary changes to make thread abort work with the new exception handling.

### CODE REVIEW

File: src/coreclr/vm/exceptionhandling.cpp [hunk 1]
```diff
@@ -7559,6 +7563,8 @@ extern "C" void * QCALLTYPE CallCatchFunclet(QCall::ObjectHandleOnStack exceptio
     GCX_COOP_NO_DTOR();
 
     Thread* pThread = GET_THREAD();
+    pThread->DecPreventAbort();
```
Comment (jkotas) [inline src/coreclr/vm/exceptionhandling.cpp:7566]: This handles exception caught by managed code. Do we need to do this when the exception is caught by the unmanaged code?

Comment (janvorli) [reply src/coreclr/vm/exceptionhandling.cpp:7566]: No, we don't
* when an exception crosses managed to native boundary, it still calls the `CallCatchFunclet` with NULL handler. That results in invoking `RealCOMPlusThrow` from the native frame that we've hit. The count is updated properly here. 
* when an exception is thrown by native code and handled by native code without ever passing through managed code, it doesn't need to fiddle with the count, as it never calls the managed EH code

**Review by jkotas — APPROVED** [review]
Thanks

**Comment by AndyAyersMS** [comment]
We are seeing some runtime failures on arm32 and x86 -- wondering if they could be related to this change?

https://dev.azure.com/dnceng-public/public/_build/results?buildId=494156&view=ms.vss-test-web.build-test-results-tab
https://dev.azure.com/dnceng-public/public/_build/results?buildId=494451&view=ms.vss-test-web.build-test-results-tab

**Comment by janvorli** [comment]
@AndyAyersMS the code in this change is never executed without the new exception handling being enabled. And it is still disabled by default.

**Comment by janvorli** [comment]
Also, it is not on any Windows x86 code path. The new exception handling is not implemented for x86 Windows

**Comment by janvorli** [comment]
But let me investigate the failures.

**Comment by BruceForstall** [comment]
Do we currently support thread abort in .NET (Core)? If not, do we plan to add (or re-add) support?

I've been seeing various comments in RyuJIT about how we do various things due to thread abort support (and especially, automatic exception re-raise after a handler is run) and I've wondered if we could remove that code to simplify generated code and the JIT itself.

**Comment by janvorli** [comment]
@BruceForstall we do. It was being enabled for internal purposes only for funceval in debugger. But about a year ago, we have enabled it for external usage using controlled execution abi: https://github.com/dotnet/runtime/pull/71661. Internally, it is still the same old thread abort, only the user interface to it is designed differently.

**Comment by janvorli** [comment]
@AndyAyersMS regarding the issue itself, the core dumps collected are not useful. The Windows x86 one kills WinDbg when I attempt to load it and the Linux arm is missing locals that would allow me to figure out why the assert fired. Do you happen to know if the issue is reproducible locally using the DOTNET_xxx settings the CI test was using?

**Comment by jkotas** [comment]
> Internally, it is still the same old thread abort, only the user interface to it is designed differently.

Thread abort in .NET Framework tried to guarantee that the thread abort won't corrupt thread or process state. I expect that most thread abort comments in the JIT are related to these guarantees.

The thread abort for funceval / controlled execution does not make these guarantees. It is best effort only.

**Comment by AndyAyersMS** [comment]
> Do you happen to know if the issue is reproducible locally using the DOTNET_xxx settings the CI test was using?

I haven't tried reproing these yet... I can give it a try later today.

**Comment by BruceForstall** [comment]
Here's one example of an optimization we always do which the comments say we can't do in runtimes with thread abort:

https://github.com/dotnet/runtime/blob/939f28ababb954ec6966cde6b2eb0b96db40cf69/src/coreclr/jit/fgehopt.cpp#L261-L277

Another issue is with automatic re-raise of thread abort exceptions: the JIT needs to insert NOPs sometimes to ensure the address used by the VM for the re-raise, which is chosen as the catch clause return address, is in the correct EH region:

https://github.com/dotnet/runtime/blob/939f28ababb954ec6966cde6b2eb0b96db40cf69/src/coreclr/jit/fgopt.cpp#L2892-L2898

I think there are other cases where NOP are inserted due to call-to-finally and similar, due to thread abort requirements.

**Comment by janvorli** [comment]
I was able to repro the issue on x86 Windows. In that case, the current thread's m_pFrame points to an address that is lower than the current SP and that fires the assert. This is likely a result of missing pop of that frame at some point. I don't know what that frame was yet, as its contents is already overwritten.

**Comment by janvorli** [comment]
It looks like for some reason, an InlinedCallFrame is not popped from the thread's m_pFrame chain. Here is a relevant part of the stack trace close before the assert. After the point shown by the stack trace, the InlinedCallFrame at 0d86c9a4 got popped by JIT_PInvokeBegin and continuing after that crashed with the assert. The InlinedCallFrame at 0d86c9f0 stayed pushed and now contained a garbage. 
```
Child SP       IP Call Site
0D86C990 69F0DD95 coreclr!JIT_PInvokeBegin + 69 at C:\git\runtime2\src\coreclr\vm\i386\PInvokeStubs.asm:70
0D86C9A4          [InlinedCallFrame: 0d86c9a4] 
0D86C9A0 684144f6 System.Private.CoreLib.dll!System.Runtime.CompilerServices.RuntimeHelpers.AllocateTypeAssociatedMemory(System.Type, Int32) + 86 [C:\git\runtime2\src\coreclr\System.Private.CoreLib\src\System\Runtime\CompilerServices\RuntimeHelpers.CoreCLR.cs @ 344]
0D86C9EC 0facbcdb Microsoft.Interop.Tests.NativeExports.dll!NativeExports.ComInterfaceGenerator.GetPoint+ImplementingObject+ABI.get_VTable() + 83 [C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 71]
0D86C9F0          [InlinedCallFrame: 0d86c9f0] 
0D86CA40 0facbc5c Microsoft.Interop.Tests.NativeExports.dll!NativeExports.ComInterfaceGenerator.GetPoint+MyComWrapper.ComputeVtables(System.Object, System.Runtime.InteropServices.CreateComInterfaceFlags, Int32 ByRef) + 108 [C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 38]
0D86CA68 6840c3f3 System.Private.CoreLib.dll!System.Runtime.InteropServices.ComWrappers.CallComputeVtables(System.Runtime.InteropServices.ComWrappersScenario, System.Runtime.InteropServices.ComWrappers, System.Object, System.Runtime.InteropServices.CreateComInterfaceFlags, Int32 ByRef) + 99 [C:\git\runtime2\src\coreclr\System.Private.CoreLib\src\System\Runtime\InteropServices\ComWrappers.cs @ 184]
0D86CA84 69F0D04B coreclr!CallDescrWorkerInternal + 52 at C:\git\runtime2\src\coreclr\vm\i386\asmhelpers.asm:520
0D86CA90 69C3D527 coreclr!CallDescrWorker + 170 at C:\git\runtime2\src\coreclr\vm\callhelpers.cpp:123
0D86CEAC 69C3D665 coreclr!CallDescrWorkerWithHandler + 222 at C:\git\runtime2\src\coreclr\vm\callhelpers.cpp:69
0D86CF24 69C3DEE1 coreclr!DispatchCallSimple + 245 at C:\git\runtime2\src\coreclr\vm\callhelpers.cpp:221
0D86CFC4 69E0A268 coreclr!`anonymous namespace'::CallComputeVTables + 1296 at C:\git\runtime2\src\coreclr\vm\interoplibinterface_comwrappers.cpp:598
0D86D13C 69E149E6 coreclr!`anonymous namespace'::TryGetOrCreateComInterfaceForObjectInternal + 1155 at C:\git\runtime2\src\coreclr\vm\interoplibinterface_comwrappers.cpp:722
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
After the CallComputeVTables returns, this frame invokes the code that asserts
--------------------------------------------------------------------
0D86D2A8 69E17086 coreclr!ComWrappers_TryGetOrCreateComInterfaceForObject + 246 at C:\git\runtime2\src\coreclr\vm\interoplibinterface_comwrappers.cpp:1489
0D86D3AC 6840C365 System_Private_CoreLib!System.Runtime.InteropServices.ComWrappers.TryGetOrCreateComInterfaceForObjectInternal + 53 at C:\git\runtime2\src\coreclr\System.Private.CoreLib\Microsoft.Interop.LibraryImportGenerator\Microsoft.Interop.LibraryImportGenerator\LibraryImports.g.cs:3055
0D86D3C8          [InlinedCallFrame: 0d86d3c8] 
0D86D3C4 6840c35e System.Private.CoreLib.dll!System.Runtime.InteropServices.ComWrappers.TryGetOrCreateComInterfaceForObjectInternal(System.Runtime.CompilerServices.ObjectHandleOnStack, Int64, System.Runtime.CompilerServices.ObjectHandleOnStack, System.Runtime.InteropServices.CreateComInterfaceFlags, IntPtr ByRef) + 46 [C:\git\runtime2\src\coreclr\System.Private.CoreLib\Microsoft.Interop.LibraryImportGenerator\Microsoft.Interop.LibraryImportGenerator\LibraryImports.g.cs @ 3055]
0D86D418 6840c315 System.Private.CoreLib.dll!System.Runtime.InteropServices.ComWrappers.TryGetOrCreateComInterfaceForObjectInternal(System.Runtime.InteropServices.ComWrappers, System.Object, System.Runtime.InteropServices.CreateComInterfaceFlags, IntPtr ByRef) + 53 [C:\git\runtime2\src\coreclr\System.Private.CoreLib\src\System\Runtime\InteropServices\ComWrappers.cs @ 154]
0D86D434 0facbb19 Microsoft.Interop.Tests.NativeExports.dll!NativeExports.ComInterfaceGenerator.GetPoint.CreatePointProvider() + 89 [C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 25]
0D86D45C 0FACB61A ComInterfaceGenerator_Tests!ComInterfaceGenerator.Tests.PreserveSigTests.CallRcwFromGeneratedComInterface() + 58
0D86D460          [InlinedCallFrame: 0d86d460] ComInterfaceGenerator.Tests.dll!ComInterfaceGenerator.Tests.PreserveSigTests.NewNativeObject()
0D86D45C 0facb61a ComInterfaceGenerator.Tests.dll!ComInterfaceGenerator.Tests.PreserveSigTests.CallRcwFromGeneratedComInterface() + 58 [C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\ComInterfaceGenerator.Tests\PreserveSigTests.cs @ 20]

```

**Comment by janvorli** [comment]
One more note - the frame that was not popped was pushed by the `CORINFO_HELP_INIT_PINVOKE_FRAME`, unlike the other that called J`IT_PInvokeBegin`.

**Comment by janvorli** [comment]
Here is the disassembly of the method that called the `CORINFO_HELP_INIT_PINVOKE_FRAME`.
```
0:011> !u 0x102cc96b
Normal JIT generated code
NativeExports.ComInterfaceGenerator.GetPoint+ImplementingObject+ABI.get_VTable()
ilAddr is 0C4E9F78 pImport is 14193350
Begin 102CC948, size 135

C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 69:
102cc948 55              push    ebp
102cc949 8bec            mov     ebp,esp
102cc94b 57              push    edi
102cc94c 56              push    esi
102cc94d 53              push    ebx
102cc94e 83ec40          sub     esp,40h
102cc951 33c9            xor     ecx,ecx
102cc953 894dec          mov     dword ptr [ebp-14h],ecx
102cc956 894de8          mov     dword ptr [ebp-18h],ecx
102cc959 894de4          mov     dword ptr [ebp-1Ch],ecx
102cc95c c745f078563412  mov     dword ptr [ebp-10h],12345678h
102cc963 8d7db8          lea     edi,[ebp-48h]
102cc966 e8896a1ef3      call    CORINFO_HELP_INIT_PINVOKE_FRAME (034b33f4)
>>> 102cc96b 8bcd            mov     ecx,ebp
102cc96d 894dcc          mov     dword ptr [ebp-34h],ecx
102cc970 f6056975341001  test    byte ptr ds:[10347569h],1
102cc977 0f84ca000000    je      Microsoft_Interop_Tests_NativeExports!NativeExports.ComInterfaceGenerator.GetPoint.ImplementingObject.ABI.get_VTable+0xff (102cca47)
102cc97d 8b0de4783410    mov     ecx,dword ptr ds:[103478E4h]
102cc983 85c9            test    ecx,ecx
102cc985 0f8598000000    jne     Microsoft_Interop_Tests_NativeExports!NativeExports.ComInterfaceGenerator.GetPoint.ImplementingObject.ABI.get_VTable+0xdb (102cca23)

C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 71:
102cc98b b914aa8a08      mov     ecx,88AAA14h (Object: System.RuntimeType)
102cc990 ba14000000      mov     edx,14h
102cc995 ff1530edcd08    call    dword ptr [CLRStub[MethodDescPrestub]@08CDED30 (08cded30)] (System.Runtime.CompilerServices.RuntimeHelpers.AllocateTypeAssociatedMemory(System.Type, Int32), mdToken: 0600682C)
102cc99b 8bf8            mov     edi,eax

C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 72:
102cc99d 8d45e4          lea     eax,[ebp-1Ch]
102cc9a0 8945dc          mov     dword ptr [ebp-24h],eax
102cc9a3 8b45dc          mov     eax,dword ptr [ebp-24h]
102cc9a6 8d55e8          lea     edx,[ebp-18h]
102cc9a9 8955d8          mov     dword ptr [ebp-28h],edx
102cc9ac 8b55d8          mov     edx,dword ptr [ebp-28h]
102cc9af 8d4dec          lea     ecx,[ebp-14h]
102cc9b2 894dd4          mov     dword ptr [ebp-2Ch],ecx
102cc9b5 50              push    eax
102cc9b6 52              push    edx
102cc9b7 8b45d4          mov     eax,dword ptr [ebp-2Ch]
102cc9ba 50              push    eax
102cc9bb c745c048e9b10b  mov     dword ptr [ebp-40h],0BB1E948h
102cc9c2 8bc4            mov     eax,esp
102cc9c4 8945c4          mov     dword ptr [ebp-3Ch],eax
102cc9c7 8d05dac92c10    lea     eax,[Microsoft_Interop_Tests_NativeExports!NativeExports.ComInterfaceGenerator.GetPoint.ImplementingObject.ABI.get_VTable+0x92 (102cc9da)]
102cc9cd 8945c8          mov     dword ptr [ebp-38h],eax
102cc9d0 c6460800        mov     byte ptr [esi+8],0
102cc9d4 ff15cce9b10b    call    dword ptr ds:[0BB1E9CCh] (System.Runtime.InteropServices.ComWrappers.<GetIUnknownImplInternal>g____PInvoke|25_0(IntPtr*, IntPtr*, IntPtr*), mdToken: 06006466)
102cc9da c6460801        mov     byte ptr [esi+8],1
102cc9de 833d6c89336a00  cmp     dword ptr [coreclr!g_TrapReturningThreads (6a33896c)],0
102cc9e5 7405            je      Microsoft_Interop_Tests_NativeExports!NativeExports.ComInterfaceGenerator.GetPoint.ImplementingObject.ABI.get_VTable+0xa4 (102cc9ec)
102cc9e7 e814f4a259      call    coreclr!JIT_RareDisableHelper (69cfbe00)
102cc9ec 33c0            xor     eax,eax
102cc9ee 8945c8          mov     dword ptr [ebp-38h],eax

C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 73:
102cc9f1 8b45ec          mov     eax,dword ptr [ebp-14h]
102cc9f4 8907            mov     dword ptr [edi],eax

C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 74:
102cc9f6 8d4704          lea     eax,[edi+4]
102cc9f9 8b4de8          mov     ecx,dword ptr [ebp-18h]
102cc9fc 8908            mov     dword ptr [eax],ecx

C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 75:
102cc9fe 8d4708          lea     eax,[edi+8]
102cca01 8b4de4          mov     ecx,dword ptr [ebp-1Ch]
102cca04 8908            mov     dword ptr [eax],ecx

C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 76:
102cca06 c7470c48063610  mov     dword ptr [edi+0Ch],offset NativeExports.ComInterfaceGenerator.GetPoint+ImplementingObject+ABI.GetPoint(ComInterfaceDispatch*) (10360648)

C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 77:
102cca0d c7471060063610  mov     dword ptr [edi+10h],offset NativeExports.ComInterfaceGenerator.GetPoint+ImplementingObject+ABI.SetPoint(ComInterfaceDispatch*, System.Drawing.Point) (10360660)

C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 78:
102cca14 f6056975341001  test    byte ptr ds:[10347569h],1
102cca1b 743e            je      Microsoft_Interop_Tests_NativeExports!NativeExports.ComInterfaceGenerator.GetPoint.ImplementingObject.ABI.get_VTable+0x113 (102cca5b)
102cca1d 893de4783410    mov     dword ptr ds:[103478E4h],edi

C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 70:
102cca23 f6056975341001  test    byte ptr ds:[10347569h],1
102cca2a 7440            je      Microsoft_Interop_Tests_NativeExports!NativeExports.ComInterfaceGenerator.GetPoint.ImplementingObject.ABI.get_VTable+0x124 (102cca6c)
102cca2c a1e4783410      mov     eax,dword ptr ds:[103478E4h]

C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 78:
102cca31 817df078563412  cmp     dword ptr [ebp-10h],12345678h
102cca38 7405            je      Microsoft_Interop_Tests_NativeExports!NativeExports.ComInterfaceGenerator.GetPoint.ImplementingObject.ABI.get_VTable+0xf7 (102cca3f)
102cca3a e86108a359      call    coreclr!JIT_FailFast (69cfd2a0)
102cca3f 8d65f4          lea     esp,[ebp-0Ch]
102cca42 5b              pop     ebx
102cca43 5e              pop     esi
102cca44 5f              pop     edi
102cca45 5d              pop     ebp
102cca46 c3              ret

C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 69:
102cca47 b918743410      mov     ecx,10347418h
102cca4c ba39010000      mov     edx,139h
102cca51 e80e691ef3      call    CORINFO_HELP_GETSHARED_NONGCSTATIC_BASE (034b3364)
102cca56 e922ffffff      jmp     Microsoft_Interop_Tests_NativeExports!NativeExports.ComInterfaceGenerator.GetPoint.ImplementingObject.ABI.get_VTable+0x35 (102cc97d)

C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 78:
102cca5b b918743410      mov     ecx,10347418h
102cca60 ba39010000      mov     edx,139h
102cca65 e8fa681ef3      call    CORINFO_HELP_GETSHARED_NONGCSTATIC_BASE (034b3364)
102cca6a ebb1            jmp     Microsoft_Interop_Tests_NativeExports!NativeExports.ComInterfaceGenerator.GetPoint.ImplementingObject.ABI.get_VTable+0xd5 (102cca1d)

C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 70:
102cca6c b918743410      mov     ecx,10347418h
102cca71 ba39010000      mov     edx,139h
102cca76 e8e9681ef3      call    CORINFO_HELP_GETSHARED_NONGCSTATIC_BASE (034b3364)
102cca7b ebaf            jmp     Microsoft_Interop_Tests_NativeExports!NativeExports.ComInterfaceGenerator.GetPoint.ImplementingObject.ABI.get_VTable+0xe4 (102cca2c)
```

**Comment by janvorli** [comment]
In this code, I can't see popping of the `InlinedCallFrame`. Looking at the disasembly without JIT stress enabled, the popping is present, as you can see in my comment below:
```
ilAddr is 0D309F78 pImport is 12AF9D68
Begin 0EB5B0D0, size ed

C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 69:
0eb5b0d0 55              push    ebp
0eb5b0d1 8bec            mov     ebp,esp
0eb5b0d3 57              push    edi
0eb5b0d4 56              push    esi
0eb5b0d5 53              push    ebx
0eb5b0d6 83ec38          sub     esp,38h
0eb5b0d9 33c9            xor     ecx,ecx
0eb5b0db 894df0          mov     dword ptr [ebp-10h],ecx
0eb5b0de 894dec          mov     dword ptr [ebp-14h],ecx
0eb5b0e1 894de8          mov     dword ptr [ebp-18h],ecx
0eb5b0e4 8d7dc0          lea     edi,[ebp-40h]
0eb5b0e7 e8088370f6      call    CORINFO_HELP_INIT_PINVOKE_FRAME (052633f4)
>>> 0eb5b0ec 8bcd            mov     ecx,ebp
0eb5b0ee 894dd4          mov     dword ptr [ebp-2Ch],ecx
0eb5b0f1 f60591f07f1001  test    byte ptr ds:[107FF091h],1
0eb5b0f8 0f84ab000000    je      Microsoft_Interop_Tests_NativeExports!NativeExports.ComInterfaceGenerator.GetPoint+ImplementingObject+ABI.get_VTable()+0xd9 (0eb5b1a9)
0eb5b0fe 833d0cf47f1000  cmp     dword ptr ds:[107FF40Ch],0
0eb5b105 0f858b000000    jne     Microsoft_Interop_Tests_NativeExports!NativeExports.ComInterfaceGenerator.GetPoint+ImplementingObject+ABI.get_VTable()+0xc6 (0eb5b196)

C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 71:
0eb5b10b b9fca5d208      mov     ecx,8D2A5FCh (Object: System.RuntimeType)
0eb5b110 ba14000000      mov     edx,14h
0eb5b115 ff1530ed1509    call    dword ptr [CLRStub[MethodDescPrestub]@0915ED30 (0915ed30)] (System.Runtime.CompilerServices.RuntimeHelpers.AllocateTypeAssociatedMemory(System.Type, Int32), mdToken: 0600682C)
0eb5b11b 8bf8            mov     edi,eax

C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 72:
0eb5b11d 8d45e8          lea     eax,[ebp-18h]
0eb5b120 8945e4          mov     dword ptr [ebp-1Ch],eax
0eb5b123 8b45e4          mov     eax,dword ptr [ebp-1Ch]
0eb5b126 8d55ec          lea     edx,[ebp-14h]
0eb5b129 8955e0          mov     dword ptr [ebp-20h],edx
0eb5b12c 8b55e0          mov     edx,dword ptr [ebp-20h]
0eb5b12f 8d4df0          lea     ecx,[ebp-10h]
0eb5b132 894ddc          mov     dword ptr [ebp-24h],ecx
0eb5b135 50              push    eax
0eb5b136 52              push    edx
0eb5b137 8b45dc          mov     eax,dword ptr [ebp-24h]
0eb5b13a 50              push    eax
0eb5b13b c745c850ace10c  mov     dword ptr [ebp-38h],0CE1AC50h
0eb5b142 8bc4            mov     eax,esp
0eb5b144 8945cc          mov     dword ptr [ebp-34h],eax
0eb5b147 8d055ab1b50e    lea     eax,[Microsoft_Interop_Tests_NativeExports!NativeExports.ComInterfaceGenerator.GetPoint+ImplementingObject+ABI.get_VTable()+0x8a (0eb5b15a)]
0eb5b14d 8945d0          mov     dword ptr [ebp-30h],eax
0eb5b150 c6460800        mov     byte ptr [esi+8],0
0eb5b154 ff15d4ace10c    call    dword ptr ds:[0CE1ACD4h] (System.Runtime.InteropServices.ComWrappers.<GetIUnknownImplInternal>g____PInvoke|25_0(IntPtr*, IntPtr*, IntPtr*), mdToken: 06006466)
0eb5b15a c6460801        mov     byte ptr [esi+8],1
0eb5b15e 833d6c89336a00  cmp     dword ptr [coreclr!g_TrapReturningThreads (6a33896c)],0
0eb5b165 7405            je      Microsoft_Interop_Tests_NativeExports!NativeExports.ComInterfaceGenerator.GetPoint+ImplementingObject+ABI.get_VTable()+0x9c (0eb5b16c)
0eb5b167 e8940c1a5b      call    coreclr!JIT_RareDisableHelper (69cfbe00)
0eb5b16c 33c0            xor     eax,eax
0eb5b16e 8945d0          mov     dword ptr [ebp-30h],eax

C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 73:
0eb5b171 8b45f0          mov     eax,dword ptr [ebp-10h]
0eb5b174 8907            mov     dword ptr [edi],eax

C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 74:
0eb5b176 8b45ec          mov     eax,dword ptr [ebp-14h]
0eb5b179 894704          mov     dword ptr [edi+4],eax

C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 75:
0eb5b17c 8b45e8          mov     eax,dword ptr [ebp-18h]
0eb5b17f 894708          mov     dword ptr [edi+8],eax

C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 76:
0eb5b182 c7470cb82fb30e  mov     dword ptr [edi+0Ch],offset NativeExports.ComInterfaceGenerator.GetPoint+ImplementingObject+ABI.GetPoint(ComInterfaceDispatch*) (0eb32fb8)

C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 77:
0eb5b189 c74710d02fb30e  mov     dword ptr [edi+10h],offset NativeExports.ComInterfaceGenerator.GetPoint+ImplementingObject+ABI.SetPoint(ComInterfaceDispatch*, System.Drawing.Point) (0eb32fd0)

C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 78:
0eb5b190 893d0cf47f10    mov     dword ptr ds:[107FF40Ch],edi

C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 70:
0eb5b196 a10cf47f10      mov     eax,dword ptr ds:[107FF40Ch]
0eb5b19b 8b4dc4          mov     ecx,dword ptr [ebp-3Ch]
0eb5b19e 894e0c          mov     dword ptr [esi+0Ch],ecx
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
This pops the InlinedCallFrame, `esi` is the current Thread pointer.
----------------------------------------------------------------
C:\git\runtime2\src\libraries\System.Runtime.InteropServices\tests\TestAssets\NativeExports\ComInterfaceGenerator\GetPoint.cs @ 78:
0eb5b1a1 8d65f4          lea     esp,[ebp-0Ch]
0eb5b1a4 5b              pop     ebx
0eb5b1a5 5e              pop     esi
0eb5b1a6 5f              pop     edi
0eb5b1a7 5d              pop     ebp
0eb5b1a8 c3              ret

```

**Comment by AndyAyersMS** [comment]
Thank Jan, opened https://github.com/dotnet/runtime/issues/95887 for this.

Looking at jit commits nothing jumps out so I'll try some bisecting I guess.

**Pull request merged**